enterprise/admin/__init__.py

```diff
@@ -135,12 +167,89 @@ class EnterpriseCustomerUserAdmin(admin.ModelAdmin):
     class Meta(object):
         model = EnterpriseCustomerUser
 
-    fields = ('user_id', 'enterprise_customer')
+    fields = (
+        'user_id',
+        'enterprise_customer',
+        'user_email',
+        'username',
+        'created',
+        'enrolled_courses',
+    )
+
+    # Only include fields that are not database-backed; DB-backed fields
+    # are dynamically set as read only.
+    readonly_fields = (
+        'user_email',
+        'username',
+        'created',
+        'enrolled_courses',
+    )
+
+    inlines = (
+        HistoricalUserDataSharingConsentAuditInlineAdmin,
+    )
+
+    def username(self, ecu):
+        """
+        Return the username for the attached user.
+        """
+        return ecu.user.username
+
+    def enrolled_courses(self, ecu):
+        """
+        Return a string represented the courses enrolled in
+        """
+        courses_string = mark_safe(self.get_enrolled_course_string(ecu))
```

Comment:
I always get concerned about exceptions when I see nested function calls -- do we need to protect against any errors here?